### PR TITLE
[master <  T0986-MG] Add parser stress test

### PR DESCRIPTION
**Pull request by gitbuda**
- [x] Simplify the test
- [x] Introduce and upgrade mgclient
- [x] Compare `mgclient` and `memgraph::communication::bolt::Client`

### CODE REVIEW

File: tests/stress/continuous_integration [hunk 1]
```diff
@@ -26,6 +26,11 @@ SMALL_DATASET = [
         "options": ["--vertex-count", "40000", "--create-pack-size", "100"],
         "timeout": 5,
     },
+    {
+        "test": "parser.cpp",
+        "options": ["--client-count", "12", "--per-client-query-count", "1000"],
+        "timeout": 5,
+    },
```
Comment (gitbuda) [inline tests/stress/continuous_integration:33]: This is the only change in this file, everything else was the pre-commit `black` hook

**Comment by gitbuda** [comment]
Running `./tests/stress/parser --client-count 8 --per-client-query-count 1000` takes:
* with `memgraph::communication::bolt::Client` 0.42s
* with `mgclient` 0.48s

which is fine in my view, we should use `mgclient` everywhere in the codebase.

File: tests/stress/parser.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  spdlog::info(fmt::format("Starting parser stress test with {} clients and {} queries per client...",
+                           FLAGS_client_count, FLAGS_per_client_query_count));
+  std::vector<std::thread> threads;
+  memgraph::utils::Timer timer;
+  for (int i = 0; i < FLAGS_client_count; ++i) {
+    threads.push_back(std::thread([]() {
+      auto client = make_client();
+      std::mt19937 generator{std::random_device{}()};
+      std::uniform_int_distribution<uint64_t> distribution{std::numeric_limits<uint64_t>::min(),
+                                                           std::numeric_limits<uint64_t>::max()};
```
Comment (jbajic) [inline tests/stress/parser.cpp:58]: Move outside of loop, or did you intentionally do this to get a different seed every time?

File: tests/stress/parser.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          LOG_FATAL("One of the parser stress test queries failed.");
+        }
+      }
+    }));
+  }
+
+  for (int i = 0; i < FLAGS_client_count; ++i) {
+    threads[i].join();
+  }
+  auto elapsed = timer.Elapsed();
```
Comment (jbajic) [inline tests/stress/parser.cpp:76]:
```suggestion
  const auto elapsed = timer.Elapsed();
```

File: tests/stress/parser.cpp [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        } catch (const std::exception &e) {
+          LOG_FATAL("One of the parser stress test queries failed.");
+        }
+      }
+    }));
+  }
+
+  for (int i = 0; i < FLAGS_client_count; ++i) {
+    threads[i].join();
+  }
```
Comment (jbajic) [inline tests/stress/parser.cpp:75]:
```suggestion
std::ranges::for_each(threads, [](auto &thread) { thread.join(); });
```

File: tests/stress/parser.cpp (hunk 3 above)
Comment (gitbuda) [reply tests/stress/parser.cpp:76]: I actually don't need that variable at all, I'll just inline the call in the next line 😄

File: tests/stress/parser.cpp (hunk 4 above)
Comment (gitbuda) [reply tests/stress/parser.cpp]: This is nice! Done!

File: tests/stress/parser.cpp (hunk 2 above)
Comment (jbajic) [reply tests/stress/parser.cpp:58]: Oh sorry didn't noticed that is in lambda while commenting

**Review by jbajic — APPROVED** [review]
Looks good!

**Pull request merged**